src/playground/sample_data_ingestion/sample_data_plugins/sample_data_plugin.py

```diff
@@ -0,0 +1,145 @@
+from datetime import datetime, timedelta
+from os import environ, path, remove
+from json import dumps, load, loads
+from shutil import copyfileobj
+from requests import request
+from base64 import b64encode
+from time import mktime
+import gzip
+
+
+"""
+Generic Plugin class: initialization is defined here
+Arguments:
+    - index_name: The name of the source index
+    - payload: The JSON body of which to call create API requests
+    - base_url: The IP address/url where the OpenSearch backend lies
+"""
+class Plugin:
+    def __init__(self, index_name, payload, base_url):
+        self.index_name = index_name
+        self.id = ""
+        self.payload = payload
+        
+        # Convert to JSON string if necessary
+        if type(self.payload) is dict:
+            self.payload = dumps(self.payload)
+        
+        # Add authentication headers
+        __auth_plain = environ.get("SAMPLE_DATA_USERNAME") + ":" + environ.get("SAMPLE_DATA_PASSWORD")
```

Comment:
Could you please provide more detail in the top for SAMPLE_DATA_USERNAME and SAMPLE_DATA_PASSWORD? if possiable to add README.md

Reply:
I changed logic so now the class uses an authentication class to build the header